spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -247,6 +250,11 @@ private Environment getRemoteEnvironment(RestTemplate restTemplate,
 				if (StringUtils.hasText(token)) {
 					headers.add(TOKEN_HEADER, token);
 				}
+				// Sending the client headers to server "headers:"
+				ArrayList<String> keyList = new ArrayList<String>(clientHeaders.keySet());
```

Comment:
This is already happening below, see `GenericRequestHeaderInterceptor`and `getSecureRestTemplate()`